lib/lotus/logger.rb

```diff
@@ -0,0 +1,82 @@
+require 'logger'
+
+module Lotus
+  # Lotus default logger
+  #
+  # Implement with the same interface of ruby std ::Logger
+  # Opt out the logdev parameter and always use STDOUT to output the log
+  #
+  # Lotus logger also has the app tag concept, which used to identify
+  # which application the log come from.
+  # Lotus Logger default come with Lotus::Application
+  # and take name of class which inherit Lotus::Application as app_tag
+  #
+  # When stand alone, Lotus Logger try to infer app tag from highest namespace
+  # When has no namespace, Lotus Logger take [Shared] as default app tag
+  #
+  # Examples
+  # #1 Logger with namespace
+  # module TestApp
+  #   class AppLogger << Lotus::Logger; end
+  #   def log
+  #     Applogger.new.info('foo')
+  #     #=> output: I, [2015-01-10T21:55:12.727259 #80487]  INFO -- [TestApp] : foo
+  #   end
+  # end
+  #
+  # #2 Logger without namespace
+  # class AppLogger < Lotus::Logger
+  # end
+  #
+  # Applogger.new.info('foo')
+  # #=> output: I, [2015-01-10T21:55:12.727259 #80487]  INFO -- [Shared] : foo
+  #
+  # @since 0.2.1
+  class Logger < ::Logger
+
+    attr_accessor :app_tag
+
+    # Override Ruby's Logger#initialize
+    #
+    # @param logdev is default to STDOUT
+    #
+    # @since 0.2.1
+    def initialize(app_tag=nil, *args)
+      super(STDOUT, *args)
+      @app_tag = app_tag
+      @formatter = Lotus::Logger::Formatter.new.tap { |f| f.app_tag = self.app_tag }
+    end
+
+    # app_tag is the identification of current app
+    # app_tag is default to use highest namespace if current namespace
+    # if app_tag is blank, lotus use default app_tag 'shared'
+    # @param logdev is default to STDOUT
+    #
+    # @since 0.2.1
+    def app_tag
+      @app_tag || _app_tag_from_namespace || _default_app_tag
+    end
+
+    class Formatter < ::Logger::Formatter
+      attr_accessor :app_tag
+
+      def call(severity, time, progname, msg)
+        raise "you must set app_tag at Logger constructor like Lotus::Logger.new('foo')" unless @app_tag
```

Comment:
@huydx Shouldn't this be moved to `Lotus::Logger#initialize`?

If this is a mandatory information, we want the application fail to start at all. With the current design, it would start but then fail as soon a logging instruction is met.

This suggested change would also eliminate the runtime overhead to check for this information every time we want to log.